src/bot/parse/parsePullRequestBotCommandLine.ts

```diff
@@ -0,0 +1,100 @@
+import assert from "assert"
+
+import { botPullRequestCommentMention, botPullRequestIgnoreCommands } from "src/bot"
+import { CancelCommand, GenericCommand, HelpCommand, ParsedCommand } from "src/bot/parse/ParsedCommand"
+import { parseVariables } from "src/bot/parse/parseVariables"
+import {
+  fetchCommandsConfiguration,
+  getDocsUrl,
+  PIPELINE_SCRIPTS_REF,
+} from "src/command-configs/fetchCommandsConfiguration"
+import { LoggerContext } from "src/logger"
+import { validateSingleShellCommand } from "src/shell"
+
+export const parsePullRequestBotCommandLine = async (
+  rawCommandLine: string,
+  ctx: LoggerContext,
+): Promise<undefined | Error | ParsedCommand> => {
+  let commandLine = rawCommandLine.trim()
+
+  // Add trailing whitespace so that bot can be differentiated from /cmd-[?]
+  if (!commandLine.startsWith(`${botPullRequestCommentMention} `)) {
+    return
+  }
+
+  // remove "bot "
+  commandLine = commandLine.slice(botPullRequestCommentMention.length).trim()
+
+  // get first word as a subcommand
+  const subcommand = commandLine.split(" ")[0]
+
+  if (!subcommand) {
+    return new Error(`Must provide a subcommand in line ${rawCommandLine}.`)
+  }
+
+  // ignore some commands
+  if (botPullRequestIgnoreCommands.includes(subcommand)) {
+    return
+  }
+
+  commandLine = commandLine.slice(subcommand.length).trim()
+
+  switch (subcommand) {
```

Comment:
moved from https://github.com/paritytech/command-bot/pull/154/files#diff-1a40bb40e6ebaacbcff8bcee3077937a09dadcd05f307213219a3173d874de4e

What changed: 
- instead of raw typed object, returning an insance of ParsedCommand, which has needed fields
- removed `-c` parsing
- added docs link, when configuration is not found